src/town_cmd.cpp

```diff
@@ -1256,6 +1256,7 @@ static bool GrowTownWithTunnel(const Town* t, const TileIndex tile, const DiagDi
 		/* Only tunnel under a mountain if the slope is continuous for at least 4 tiles. We want tunneling to be a last resort for large hills. */
 		TileIndex slope_tile = tile;
 		for (uint8 tiles = 0; tiles < 4; tiles++) {
+			if (!IsValidTile(slope_tile)) return false;
 			slope = GetTileSlope(slope_tile);
 			if (slope != InclinedSlope(tunnel_dir) && !IsSteepSlope(slope) && !IsSlopeWithOneCornerRaised(slope)) return false;
 			slope_tile += delta;
```

Comment:
shouldn't this be TILE_ADD as well?

Reply:
seems fine to me, what do you mean?

Reply:
I mean there is a macro for adding TileIndexDiff to TileIndex instead of just +=, might as well use it

Reply:
Yes, there is `TileAddByDiagDir()`. I took the `+= delta` method from GrowTownWithBridge(). I'm not opposed to changing both to the macro, but I imagine that should be a separate codechange PR.

Or are you saying that `IsValidTile()` should check the next tile, instead of the current tile?